dlp_payload_log.go

```diff
@@ -0,0 +1,73 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+type DLPPayloadLogSettings struct {
+	PublicKey string `json:"public_key,omitempty"`
+
+	// Only present in responses
+	UpdatedAt *time.Time `json:"updated_at,omitempty"`
+}
+
+type GetDLPPayloadLogSettingsParams struct{}
+
+type DLPPayloadLogSettingsResponse struct {
+	Success  bool                  `json:"success"`
+	Errors   []string              `json:"errors"`
+	Messages []string              `json:"messages"`
+	Result   DLPPayloadLogSettings `json:"result"`
+}
+
+// GetDLPPayloadLogSettings gets the current DLP payload logging settings.
+//
+// API reference: https://api.cloudflare.com/#dlp-payload-log-settings-get-settings
+func (api *API) GetDLPPayloadLogSettings(ctx context.Context, rc *ResourceContainer, params GetDLPPayloadLogSettingsParams) (DLPPayloadLogSettings, error) {
+	if rc.Identifier == "" {
+		return DLPPayloadLogSettings{}, ErrMissingResourceIdentifier
+	}
+
+	uri := buildURI(fmt.Sprintf("/%s/%s/dlp/payload_log", rc.Level, rc.Identifier), nil)
+
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return DLPPayloadLogSettings{}, err
+	}
+
+	var dlpPayloadLogSettingsResponse DLPPayloadLogSettingsResponse
+	err = json.Unmarshal(res, &dlpPayloadLogSettingsResponse)
+	if err != nil {
+		return DLPPayloadLogSettings{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+
+	return dlpPayloadLogSettingsResponse.Result, nil
+}
+
+// PutDLPPayloadLogSettings sets thu cerrent DLP payload logging settings to new values.
+//
+// API reference: https://api.cloudflare.com/#dlp-payload-log-settings-update-settings
+func (api *API) PutDLPPayloadLogSettings(ctx context.Context, rc *ResourceContainer, settings DLPPayloadLogSettings) (DLPPayloadLogSettings, error) {
```

Comment:
generally, we have been using `Update...` for these sort of methods. can we update that here?